src/concrete/ml/sklearn/base.py

```diff
@@ -1279,13 +1287,19 @@ def __init_subclass__(cls):
                 _TREE_MODELS.add(cls)
                 _ALL_SKLEARN_MODELS.add(cls)
 
-    def __init__(self, n_bits: int):
+    def __init__(self, n_bits: Union[int, Dict[str, int]]):
         """Initialize the TreeBasedEstimatorMixin.
 
         Args:
-            n_bits (int): The number of bits used for quantization.
+            n_bits (int, Dict[str, int]): Number of bits to quantize the model. If an int is passed
+                for n_bits, the value will be used for quantizing inputs and leaves. If a dict is
+                passed, then it should contain "op_inputs" and "op_leaves" as keys with
+                corresponding number of quantization bits so that:
+                    - op_inputs : number of bits to quantize the input values
+                    - op_leaves: number of bits to quantize the leaves
```

Comment:
same, this docstring should match the one you used for the `get_n_bits_dict_trees` (indicate what is mandatory or optional, with default value if optional) !